io.openems.edge.pvinverter.sunspec/src/io/openems/edge/pvinverter/sunspec/AbstractSunSpecPvInverter.java

```diff
@@ -110,13 +115,28 @@ protected void deactivate() {
 	public void handleEvent(Event event) {
 		if (!this.isEnabled() || !this.isSunSpecInitializationCompleted()) {
 			this.channel(SunSpecPvInverter.ChannelId.PV_LIMIT_FAILED).setNextValue(false);
+			this.channel(SunSpecPvInverter.ChannelId.READ_ONLY_MODE_PV_LIMIT_FAILED).setNextValue(false);
 			return;
 		}
 
 		switch (event.getTopic()) {
 		case EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE:
+			// Get ActivePowerLimit that should be applied
+			IntegerWriteChannel activePowerLimitChannel = (IntegerWriteChannel) this
```

Comment:
either use type or cast one would be enough :)

Reply:
Indeed... 🤦‍♂️